ios/browser/api/ledger/brave_ledger.mm

```diff
@@ -442,48 +442,57 @@ - (void)createWallet:(void (^)(NSError* _Nullable))completion {
   //   malformed data
   //   - REGISTRATION_VERIFICATION_FAILED: Missing master user token
   self.initializingWallet = YES;
-  ledger->CreateRewardsWallet(base::BindOnce(^(ledger::mojom::Result result) {
-    const auto strongSelf = weakSelf;
-    if (!strongSelf) {
-      return;
-    }
-    NSError* error = nil;
-    if (result != ledger::mojom::Result::LEDGER_OK) {
-      std::map<ledger::mojom::Result, std::string> errorDescriptions{
-          {ledger::mojom::Result::LEDGER_ERROR,
-           "The wallet was already initialized"},
-          {ledger::mojom::Result::BAD_REGISTRATION_RESPONSE,
-           "Request credentials call failure or malformed data"},
-          {ledger::mojom::Result::REGISTRATION_VERIFICATION_FAILED,
-           "Missing master user token from registered persona"},
-      };
-      NSDictionary* userInfo = @{};
-      const auto description =
-          errorDescriptions[static_cast<ledger::mojom::Result>(result)];
-      if (description.length() > 0) {
-        userInfo =
-            @{NSLocalizedDescriptionKey : base::SysUTF8ToNSString(description)};
-      }
-      error = [NSError errorWithDomain:BraveLedgerErrorDomain
-                                  code:static_cast<NSInteger>(result)
-                              userInfo:userInfo];
-    }
-
-    [strongSelf startNotificationTimers];
-    strongSelf.initializingWallet = NO;
+  ledger->CreateRewardsWallet(
+      "",
+      base::BindOnce(^(ledger::mojom::CreateRewardsWalletResult create_result) {
+        const auto strongSelf = weakSelf;
+        if (!strongSelf) {
+          return;
+        }
 
-    dispatch_async(dispatch_get_main_queue(), ^{
-      if (completion) {
-        completion(error);
-      }
+        ledger::mojom::Result result =
+            create_result == ledger::mojom::CreateRewardsWalletResult::kSuccess
+                ? ledger::mojom::Result::LEDGER_OK
+                : ledger::mojom::Result::LEDGER_ERROR;
 
-      for (BraveLedgerObserver* observer in [strongSelf.observers copy]) {
-        if (observer.walletInitalized) {
-          observer.walletInitalized(static_cast<LedgerResult>(result));
+        NSError* error = nil;
+        if (result != ledger::mojom::Result::LEDGER_OK) {
+          std::map<ledger::mojom::Result, std::string> errorDescriptions{
+              {ledger::mojom::Result::LEDGER_ERROR,
+               "The wallet was already initialized"},
+              {ledger::mojom::Result::BAD_REGISTRATION_RESPONSE,
+               "Request credentials call failure or malformed data"},
+              {ledger::mojom::Result::REGISTRATION_VERIFICATION_FAILED,
+               "Missing master user token from registered persona"},
```

Comment:
Technically these entries are no longer needed as we don't map to these above (only `LEDGER_ERROR`)

Reply:
Follow-up issue filed at https://github.com/brave/brave-browser/issues/25817